### PR TITLE
#2919 Checking that the quantity total picked for a sales order line …

### PR DESCRIPTION
**Pull request by e-Evolution**
…does not exceed the amount ordered https://github.com/adempiere/adempiere/issues/2919

### CODE REVIEW

**Comment by marcalwestf** [comment]
just one question @e-Evolution : I understand that you consider the total picked quantity, but I wonder why you do not consider the already delivered quantities for the order line....

Perhaps I am missing something...

**Comment by yamelsenih** [comment]
Hi @e-Evolution @marcalwestf I am waiting for resolve #2925 for make a complete test for it, I have this implemented for may clients and is ok but I want to complete test with changes for Generate Outbound from Sales Order.

Please @e-Evolution complete changes requested for begin a complete test.

Beste regard

File: org.eevolution.warehouse/src/main/java/base/org/eevolution/process/GenerateShipmentOutBound.java [hunk 1]
```diff
@@ -110,7 +111,7 @@ private void createShipment(MWMInOutBoundLine outboundLine) {
         // Generate Shipment based on Outbound Order
         if (outboundLine.getC_OrderLine_ID() > 0) {
             MOrderLine orderLine = outboundLine.getOrderLine();
-            if (orderLine.getQtyOrdered().subtract(orderLine.getQtyDelivered()).subtract(outboundLine.getPickedQty()).signum() <= 0 && !isIncludeNotAvailable())
+            if (orderLine.getQtyOrdered().subtract(getQtyTotalPicked(outboundLine.getC_OrderLine_ID())).signum() < 0 && !isIncludeNotAvailable())
```
Comment (yamelsenih) [inline org.eevolution.warehouse/src/main/java/base/org/eevolution/process/GenerateShipmentOutBound.java]: Hi @e-Evolution you should validate for delivered quantity also, see:

```Java
if (orderLine.getQtyOrdered().subtract(orderLine.getQtyDelivered()).subtract(outboundLine.getQtyTotalPicked(outboundLine.getC_OrderLine_ID())).signum() < 0 && !isIncludeNotAvailable())
				return;
```
Best regards

**Comment by marcalwestf** [comment]
@yamelsenih  , @e-Evolution any news this PR regarding?

**Comment by e-Evolution** [comment]
fixed here  #3338

**Pull request closed without merging**